From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: kmsg: add descriptions for new messages
References: bnc#702651,LTC#72829
Patch-mainline: No

Description: This adds the kmsg descriptions for printk messages introduced
             by new System z features.

Acked-by: John Jolly <jjolly@suse.de>
---
 Documentation/kmsg/s390/aes_s390  |   15 +++
 Documentation/kmsg/s390/cio       |   27 +++++++
 Documentation/kmsg/s390/dasd      |   40 +++++++++-
 Documentation/kmsg/s390/dasd-eckd |  143 +++++++++++++++++++++++++++++++++++++-
 Documentation/kmsg/s390/dasd-fba  |    7 +
 Documentation/kmsg/s390/lcs       |    1 
 Documentation/kmsg/s390/qeth      |   56 ++++++++++----
 7 files changed, 267 insertions(+), 22 deletions(-)

--- a/Documentation/kmsg/s390/aes_s390
+++ b/Documentation/kmsg/s390/aes_s390
@@ -1,4 +1,19 @@
 /*?
+ * Text: "Allocating XTS fallback algorithm %s failed\n"
+ * Severity: Error
+ * Parameter:
+ *   @1: algorithm name
+ * Description:
+ * The aes_s390 module failed to allocate a software fallback for the AES
+ * modes that are not supported by the hardware. A possible reason for this
+ * problem is that the aes_generic module that provides the fallback
+ * algorithms is not available.
+ * User action:
+ * Ensure that the aes_generic module is available and loaded and reload
+ * the aes_s390 module.
+ */
+
+/*?
  * Text: "Allocating AES fallback algorithm %s failed\n"
  * Severity: Error
  * Parameter:
--- a/Documentation/kmsg/s390/cio
+++ b/Documentation/kmsg/s390/cio
@@ -143,3 +143,30 @@
  * User action:
  * None.
  */
+
+/*?
+ * Text: "Logging for subchannel 0.%x.%04x failed with errno=%d\n"
+ * Severity: Warning
+ * Parameter:
+ *   @1: subchannel set ID
+ *   @2: subchannel number
+ *   @3: errno
+ * Description:
+ * Capturing model-dependent logs and traces could not be triggered for the
+ * specified subchannel.
+ * User action:
+ * See the errno man page to find out what caused the problem.
+ */
+
+/*?
+ * Text: "Logging for subchannel 0.%x.%04x was triggered\n"
+ * Severity: Notice
+ * Parameter:
+ *   @1: subchannel set ID
+ *   @2: subchannel number
+ * Description:
+ * Model-dependent logs and traces may be captured for the specified
+ * subchannel.
+ * User action:
+ * None.
+ */
--- a/Documentation/kmsg/s390/dasd
+++ b/Documentation/kmsg/s390/dasd
@@ -135,28 +135,30 @@
  */
 
 /*?
- * Text: "%s: cqr %p timed out (%is), %i retries remaining\n"
+ * Text: "%s: cqr %p timed out (%lds), %i retries remaining\n"
  * Severity: Error
  * Parameter:
  *   @1: bus ID of the DASD
- *   @2: pointer to request
+ *   @2: request
  *   @3: timeout value
  *   @4: number of retries left
  * Description:
- * One try of the error recovery procedure (ERP) for the channel queued request
+ * A try of the error recovery procedure (ERP) for the channel queued request
  * (cqr) timed out and failed to recover the error. ERP continues for the DASD.
  * User action:
  * Ignore this message if it occurs infrequently and if the recovery succeeds
  * during one of the retries. If this error persists, check for related
  * previous error messages and report the problem to your support organization.
+ *
+ * The timeout can be changed by writing a new value to the sysfs 'expires' attribute of the DASD. The value specifies the timeout in seconds.
  */
 
 /*?
- * Text: "%s: cqr %p timed out (%is) but cannot be ended, retrying in 5 s\n"
+ * Text: "%s: cqr %p timed out (%lds) but cannot be ended, retrying in 5 s\n"
  * Severity: Error
  * Parameter:
  *   @1: bus ID of the DASD
- *   @2: pointer to request
+ *   @2: request
  *   @3: timeout value
  * Description:
  * A try of the error recovery procedure (ERP) for the channel queued request
@@ -167,6 +169,8 @@
  * Ignore this message if it occurs infrequently and if the recovery succeeds
  * during one of the retries. If this error persists, check for related
  * previous error messages and report the problem to your support organization.
+ *
+ * The timeout can be changed by writing a new value to the sysfs 'expires' attribute of the DASD. The value specifies the timeout in seconds.
  */
 
 /*?
@@ -465,6 +469,32 @@
  * Note the reason code and contact your support organization.
 */
 
+/*?
+ * Text: "%s: No operational channel path is left for the device\n"
+ * Severity: Warning
+ * Parameter:
+ *   @1: bus ID of the DASD
+ * Description:
+ * All channel paths to the device have become non-operational. The DASD
+ * device driver suspends I/O operations and queues I/O requests for this
+ * device until at least one channel path becomes operational again.
+ * User action:
+ * Ensure that each channel path to the device has been set up correctly
+ * and that the related physical cable connections are in place.
+ */
+
+/*?
+ * Text: "%s: A channel path to the device has become operational\n"
+ * Severity: Informational
+ * Parameter:
+ *   @1: bus ID of the DASD
+ * Description:
+ * At least one channel path of this device has become operational again.
+ * The DASD device driver resumes I/O operations to the device and processes
+ * the I/O requests that were queued while there was no operational channel path.
+ * User action:
+ * None.
+ */
 
 ------------------------------------------------------------------------------------
 /* dasd_diag */
--- a/Documentation/kmsg/s390/dasd-eckd
+++ b/Documentation/kmsg/s390/dasd-eckd
@@ -115,7 +115,7 @@
  */
 
 /*?
- * Text: "%s: New DASD %04X/%02X (CU %04X/%02X) with %d cylinders, %d heads, %d sectors\n"
+ * Text: "%s: New DASD %04X/%02X (CU %04X/%02X) with %d cylinders, %d heads, %d sectors%s\n"
  * Severity: Informational
  * Parameter:
  *   @1: bus ID of the DASD
@@ -126,8 +126,13 @@
  *   @6: number of cylinders
  *   @7: tracks per cylinder
  *   @8: sectors per track
+ *   @9: access mode
  * Description:
  * A DASD with the shown characteristics has been set online.
+ * If the DASD is configured as read-only to the real or virtual hardware,
+ * the message includes an indication of this hardware access mode. The
+ * hardware access mode is independent from the 'readonly' attribute of
+ * the device in sysfs.
  * User action:
  * None.
  */
@@ -1899,3 +1904,139 @@
  * DASD, for example, with dasdfmt.
  * ATTENTION: Formatting irreversibly destroys all data on the DASD.
  */
+
+/*?
+ * Text: "%s: An I/O request was rejected because writing is inhibited\n"
+ * Severity: Error
+ * Parameter:
+ *   @1: bus ID of the DASD
+ * Description:
+ * An I/O request was returned with an error indication of 'command reject'
+ * and 'write inhibited'. The most likely reason for this error is a
+ * failed write request to a device that was attached as read-only in z/VM.
+ * User action:
+ * Set the device offline, ensure that the device is configured correctly in
+ * z/VM, then set the device online again.
+ */
+
+/*?
+ * Text: "%s: An Alias device was reassigned to a new base device with UID: %s\n"
+ * Severity: Informational
+ * Parameter:
+ *   @1: bus ID of the alias
+ *   @2: UID of new base device
+ * Description:
+ * The alias device with the indicated bus ID has been reassigned. The UID of the new base device is shown in the message.
+ * User action:
+ * None.
+ */
+
+/*?
+ * Text: "%s: Detecting the maximum supported data size for zHPF requests failed\n"
+ * Severity: Warning
+ * Parameter:
+ *   @1: bus ID of the DASD
+ * Description:
+ * High Performance FICON (zHPF) requests are limited to a hardware-dependent
+ * maximum data size. The DASD device driver failed to detect this size and zHPF
+ * is not available for this device.
+ * User action:
+ * Set the device offline and online again. If this problem persists, gather
+ * Linux debug data and report the problem to your support organization.
+ */
+
+/*?
+ * Text: "%s: Reading device feature codes failed (rc=%d) for new path %x\n"
+ * Severity: Warning
+ * Parameter:
+ *   @1: bus ID of the DASD
+ *   @2: return code
+ *   @3: path mask
+ * Description:
+ * A new path has been made available to the a device.
+ * A command to read the device feature codes on this device returned an error.
+ * The new path will not be used for I/O.
+ * User action:
+ * Set the new path offline and online again to repeat the path verification.
+ * Alternatively, set the device offline and online again to
+ * verify all available paths for this device.
+ * If this problem persists, gather Linux debug data and report the problem
+ * to your support organization.
+ */
+
+/*?
+ * Text: "%s: Detecting the maximum data size for zHPF requests failed (rc=%d) for a new path %x\n"
+ * Severity: Warning
+ * Parameter:
+ *   @1: bus ID of the DASD
+ *   @2: return code
+ *   @3: path mask
+ * Description:
+ * High Performance FICON (zHPF) requests are limited to a hardware-dependent
+ * maximum data size. A command to detect this size for
+ * a new path returned an error. The new path will not be used for I/O.
+ * User action:
+ * Set the new path offline and online again to repeat the path verification.
+ * Alternatively, set the device offline and online again to
+ * verify all available paths for this device.
+ * If this problem persists, gather Linux debug data and report the problem
+ * to your support organization.
+ */
+
+/*?
+ * Text: "%s: The maximum data size for zHPF requests %u on a new path %x is below the active maximum %u\n"
+ * Severity: Warning
+ * Parameter:
+ *   @1: bus ID of the DASD
+ *   @2: size in bytes
+ *   @3: path mask
+ *   @4: size in bytes
+ * Description:
+ * High Performance FICON (zHPF) requests are limited to a hardware-dependent
+ * maximum data size. The maximum of the new path is below
+ * the previously established common maximum for the
+ * existing paths for this device. This could cause requests on the new
+ * path to fail. The new path will not be used for I/O.
+ * User action:
+ * Set the device offline and online again to establish a new common maximum
+ * data size for the device.
+ */
+
+/*?
+ * Text: "%s: The device reservation was lost\n"
+ * Severity: Error
+ * Parameter:
+ *   @1: bus ID of the DASD
+ * Description:
+ * This Linux instance has lost its reservation of the device to another
+ * operating system instance. Depending on the reservation policy for the
+ * device, I/O might be blocked until the other operating system instance
+ * surrenders the reservation or all I/O requests might fail until the
+ * device is reset.
+ * User action:
+ * None, if this situation is handled by system automation software.
+ * If this situation is not handled by automation, check the
+ * last_known_reservation_state attribute of the device in sysfs.
+ * If the value is 'lost', verify that the device is no longer reserved
+ * by another operating system instance, then set the device offline and
+ * online again. For any other value of the last_known_reservation_state
+ * no action is required. I/O will resume when the device reservation is
+ * surrendered by the other operating system instance.
+ */
+
+/*?
+ * Text: "%s: DASD raw track access not available\n"
+ * Severity: Error
+ * Parameter:
+ *   @1: bus ID of the DASD
+ * Description:
+ * The DASD cannot be accessed in raw-track access mode because the storage
+ * server does not have all required features for this access mode.
+ * In raw-track access mode, the DASD device driver accesses complete ECKD
+ * tracks.
+ * By default, the DASD device driver accesses only the data fields of ECKD
+ * devices and omits the count and key data fields.
+ * User action:
+ * Ensure that the raw_track_access sysfs attribute of the DASD has the value
+ * 0 to access the device in default ECKD mode.
+ */
--- a/Documentation/kmsg/s390/dasd-fba
+++ b/Documentation/kmsg/s390/dasd-fba
@@ -1,6 +1,6 @@
 
 /*?
- * Text: "%s: New FBA DASD %04X/%02X (CU %04X/%02X) with %d MB and %d B/blk\n"
+ * Text: "%s: New FBA DASD %04X/%02X (CU %04X/%02X) with %d MB and %d B/blk%s\n"
  * Severity: Informational
  * Parameter:
  *   @1: bus ID of the DASD
@@ -10,8 +10,13 @@
  *   @5: control unit model
  *   @6: size
  *   @7: bytes per block
+ *   @8: access mode
  * Description:
  * A DASD with the shown characteristics has been set online.
+ * If the DASD is configured as read-only to the real or virtual hardware,
+ * the message includes an indication of this hardware access mode. The
+ * hardware access mode is independent from the 'readonly' attribute of
+ * the device in sysfs.
  * User action:
  * None.
  */
--- a/Documentation/kmsg/s390/lcs
+++ b/Documentation/kmsg/s390/lcs
@@ -159,3 +159,4 @@
 /*? Text: "Initialization failed\n" */
 /*? Text: "Terminating lcs module.\n" */
 /*? Text: "Device %s could not be recovered!\n" */
+/*? Text: "%s: The lcs device driver failed to recover the device\n" */
--- a/Documentation/kmsg/s390/qeth
+++ b/Documentation/kmsg/s390/qeth
@@ -469,16 +469,11 @@
  */
 
 /*?
- * Text: "%s: MAC address %2.2x:%2.2x:%2.2x:%2.2x:%2.2x:%2.2x already exists\n"
+ * Text: "%s: MAC address %pM already exists\n"
  * Severity: Warning
  * Parameter:
  *   @1: bus ID of the qeth device
- *   @2: first token of the MAC-address
- *   @3: second token of the MAC-address
- *   @4: third token of the MAC-address
- *   @5: fourth token of the MAC-address
- *   @6: fifth token of the MAC-address
- *   @7: sixth token of the MAC-address
+ *   @2: MAC-address
  * Description:
  * Setting the MAC address for the qeth device fails, because this
  * MAC address is already defined on the OSA CHPID.
@@ -487,16 +482,11 @@
  */
 
 /*?
- * Text: "%s: MAC address %2.2x:%2.2x:%2.2x:%2.2x:%2.2x:%2.2x is not authorized\n"
+ * Text: "%s: MAC address %pM is not authorized\n"
  * Severity: Warning
  * Parameter:
  *   @1: bus ID of the qeth device
- *   @2: first token of the MAC-address
- *   @3: second token of the MAC-address
- *   @4: third token of the MAC-address
- *   @5: fourth token of the MAC-address
- *   @6: fifth token of the MAC-address
- *   @7: sixth token of the MAC-address
+ *   @2: MAC-address
  * Description:
  * This qeth device is a virtual network interface card (NIC), to which z/VM
  * has already assigned a MAC address. z/VM MAC address verification does
@@ -565,6 +555,41 @@
  */
 
 /*?
+ * Text: "%s: Enabling HW TX checksumming for %s failed, using SW TX checksumming\n"
+ * Severity: Warning
+ * Parameter:
+ *   @1: bus ID of the qeth device
+ *   @2: network interface name
+ * Description:
+ * The network adapter supports hardware checksumming for outgoing IP packages
+ * but the qeth device driver could not enable hardware TX checksumming on the
+ * adapter. The qeth device driver continues to use software checksumming for
+ * outgoing IP packages.
+ * User action:
+ * None if you do not require hardware checksumming for outgoing network
+ * traffic. If you want to enable hardware checksumming, ungroup and regroup
+ * the subchannel triplet of the device. If this does not resolve the problem,
+ * reboot Linux. If the problem persists, gather Linux debug data and report
+ * the problem to your support organization.
+ */
+
+/*?
+ * Text: "%s: A connection could not be established because of an OLM limit\n"
+ * Severity: Error
+ * Parameter:
+ *   @1: bus ID of the qeth device
+ * Description:
+ * z/OS has activated Optimized Latency Mode (OLM) for a connection through an OSA Express3 adapter.
+ * This reduces the maximum number of concurrent connections per physical port for shared adapters.
+ * The new connection would exceed the maximum. Linux cannot establish further connections using
+ * this adapter.
+ * User action:
+ * If possible, deactivate an existing connection that uses this adapter and try again to establish
+ * the new connection. If you cannot free an existing connection, use a different adapter for the
+ * new connection.
+ */
+
+/*?
  * Text: "%s: Setting the device online failed because of insufficient authorization\n"
  * Severity: Error
  * Parameter:
@@ -586,7 +611,7 @@
 /*? Text: "%s: Priority Queueing not supported\n" */
 /*? Text: "%s: sense data available. cstat 0x%X dstat 0x%X\n" */
 /*? Text: "loading core functions\n" */
-/*? Text: "%s: MAC address %2.2x:%2.2x:%2.2x:%2.2x:%2.2x:%2.2x successfully registered on device %s\n" */
+/*? Text: "%s: MAC address %pM successfully registered on device %s\n" */
 /*? Text: "%s: Device successfully recovered!\n" */
 /*? Text: "register layer 2 discipline\n" */
 /*? Text: "unregister layer 2 discipline\n" */
@@ -616,4 +641,5 @@
 /*? Text: "%s: Adapter does not support QDIO data connection isolation\n" */
 /*? Text: "%s: Adapter is dedicated. QDIO data connection isolation not supported\n" */
 /*? Text: "%s: TSO does not permit QDIO data connection isolation\n" */
+/*? Text: "%s: HW TX Checksumming enabled\n" */
 
